### PR TITLE
[storage] Do not return error in RemoveAllCertsInfo if no entries exist

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

**Review by mykola-kobets-epam — APPROVED** [review]

**Review by MykolaSolyanko — APPROVED** [review]

File: src/communication/vchannel.cpp [hunk 1]
```diff
@@ -53,18 +53,24 @@ aos::Error VChannel::Close()
...
-    if (!err.IsNone()) {
-        return AOS_ERROR_WRAP(err);
-    }
+    size_t read = 0;
 
-    auto ret = vch_read(&mReadHandle, data.Get(), size);
-    if (ret < 0) {
-        return AOS_ERROR_WRAP(ret);
+    while (read < size) {
+        auto err = data.Resize(size);
```
Comment (mykola-kobets-epam) [inline src/communication/vchannel.cpp:59]: this can be done outside of loop

File: src/communication/vchannel.cpp [hunk 2]
```diff
@@ -77,12 +83,18 @@ aos::Error VChannel::Read(aos::Array<uint8_t>& data, size_t size)
...
+        auto ret = vch_write(&mWriteHandle, data.Get() + written, data.Size() - written);
+        if (ret < 0) {
+            return AOS_ERROR_WRAP(ret);
+        }
+
+        written += ret;
     }
 
-    if (static_cast<size_t>(ret) != data.Size()) {
+    if (written != data.Size()) {
```
Comment (mykola-kobets-epam) [inline src/communication/vchannel.cpp:97]: that's mostly a check for implementation error: written data should not exceed size.
in this case assert is better.

File: src/communication/vchannel.cpp [hunk 3]
```diff
@@ -53,18 +53,24 @@ aos::Error VChannel::Close()
...
+        if (ret < 0) {
+            return AOS_ERROR_WRAP(ret);
+        }
+
+        read += ret;
     }
 
-    if (static_cast<size_t>(ret) != size) {
-        err = data.Resize(ret);
+    if (read != size) {
```
Comment (mykola-kobets-epam) [inline src/communication/vchannel.cpp:72]: not sure this check is required anymore.

**Review by mykola-kobets-epam — APPROVED** [review]

**Review by MykolaSolyanko — APPROVED** [review]

File: src/communication/vchannel.cpp (hunk 2 above)
Comment (al1img) [reply src/communication/vchannel.cpp:97]: It could exceeds runtime.

File: src/communication/vchannel.cpp (hunk 3 above)
Comment (al1img) [reply src/communication/vchannel.cpp:72]: I would like to return actual amount of data.

File: src/communication/vchannel.cpp (hunk 1 above)
Comment (al1img) [reply src/communication/vchannel.cpp]: done

File: src/communication/vchannel.cpp [hunk 4]
```diff
@@ -53,18 +53,24 @@ aos::Error VChannel::Close()
...
+        err = data.Resize(read);
```
Comment (mykola-kobets-epam) [inline src/communication/vchannel.cpp:73]: increasing in size will 0 additional elements. Decreasing is not possible according to while loop condition

File: src/communication/vchannel.cpp (hunk 2 above)
Comment (mykola-kobets-epam) [reply src/communication/vchannel.cpp:97]: ok

**Review by mykola-kobets-epam — APPROVED** [review]

**Review by mykola-kobets-epam — APPROVED** [review]

**Review by mykola-kobets-epam — APPROVED** [review]

**Pull request merged**